synapse/lib/ast.py

```diff
@@ -1097,6 +1097,10 @@ class SetItemOper(Oper):
     '''
     async def run(self, runt, genr):
 
+        if runt.readonly:
+            mesg = 'Storm runtime is in readonly mode, cannot create or edit nodes and other graph data.'
+            raise self.addExcInfo(s_exc.IsReadOnly(mesg=mesg))
```

Comment:
This breaks use which does not modify data:

```
$d=({"k":"v"}) $d.hehe=haha $lib.pprint($d)
```

Can we check the `stor` function readonly markings in `setitem()` implementation?